### DISCUSSION TITLE
Conditional rendering RainbowKit hooks

### DISCUSSION THREAD

**Discussion by JasperAlexander** [rainbow-me/rainbowkit#588] [Q&A] [answered]
Conditional rendering of the RainbowKit hooks using the conditional operator (?) is not working. How to solve this? Every time I refresh my page I'm getting the following react hydration errors.

```
Error: Text content does not match server-rendered HTML.
See more info here: https://nextjs.org/docs/messages/react-hydration-error

Error: There was an error while hydrating. Because the error happened outside of a Suspense boundary, the entire root will switch to client rendering.
```

This is my code. I want to only show the ENS name if it is available.
```
import type { NextPage } from 'next'
import { useAccount, useEnsName } from 'wagmi'

const Test: NextPage = () => {
    const { address } = useAccount()
    const { data: EnsName } = useEnsName({
        address: address,
        chainId: 1337 // My local chain
     })

     return (
          <main>
                <h2>
                     {typeof window !== 'undefined' 
                     ? EnsName // Removing this condition works, however, I want to check if the ENS name is available
                          ? EnsName 
                          : address 
                     : '' // Loading here
                     }
                </h2>
          </main>
     )
}
```

**Comment by markdalgleish** [comment, marked as answer]
You can't conditionally render based on the presence of `window` when doing SSR or you'll get these sorts of hydration errors. You can delay rendering until after hydration by setting a `loading` state variable to `false` inside a `useEffect`, then use the loading state for your condition.

First-class SSR support for wagmi is currently in progress which will make this sort of thing a lot easier: https://github.com/tmm/wagmi/pull/689
